### PR TITLE
Fix Looker Classifications handing

### LINKED ISSUE #5317: On the Demo Site, clicking `label tags` in the BDD example crashes the web app
### Describe the problem

When trying the BDD example and clicking on `label tags`, I see this in the browser

```
TypeError: Cannot use 'in' operator to search for 'mask' in undefined
Trace
TypeError: Cannot use 'in' operator to search for 'mask' in undefined
    at i.value (https://try.fiftyone.ai/_next/static/chunks/pages/_app-d66b0f27d628c577.js:302:1001933)
    at i.value (https://try.fiftyone.ai/_next/static/chunks/pages/_app-d66b0f27d628c577.js:302:1055460)
    at https://try.fiftyone.ai/_next/static/chunks/819.1518d7fb2196dea0.js:1:317914
    at e.value (https://try.fiftyone.ai/_next/static/chunks/7420.9da7a9f1f16a156b.js:1:482745)
    at e.value (https://try.fiftyone.ai/_next/static/chunks/7420.9da7a9f1f16a156b.js:1:491641)
    at n.value (https://try.fiftyone.ai/_next/static/chunks/7420.9da7a9f1f16a156b.js:1:468626)
    at https://try.fiftyone.ai/_next/static/chunks/819.1518d7fb2196dea0.js:1:317836
    at https://try.fiftyone.ai/_next/static/chunks/pages/_app-d66b0f27d628c577.js:302:1448187
    at https://try.fiftyone.ai/_next/static/chunks/819.1518d7fb2196dea0.js:1:317792
    at u2 (https://try.fiftyone.ai/_next/static/chunks/framework-0183e6d1d8e5eebe.js:9:92097)
```

### Code to reproduce issue

This is on the demo website, navigate to https://try.fiftyone.ai/datasets/try-bdd/samples and click `label tags` in the left navbar.

Screenshot 2024-12-26 at 10 41 18

### System information

This occurred in both Safari and Brave browsers

-   **OS Platform and Distribution** macOS 15.0.1:
-   **Python version** (`python --version`): Python 3.13.0
-   **FiftyOne version** (`fiftyone --version`): n/a
-   **FiftyOne installed from** (pip or source): n/a

### Other info/logs

<details><summary>console output from Brave Browser</summary

```

framework-0183e6d1d8e5eebe.js:9 TypeError: Cannot use 'in' operator to search for 'mask' in undefined
    at i.value (_app-d66b0f27d628c577.js:302:1001933)
    at i.value (_app-d66b0f27d628c577.js:302:1055460)
    at 819.1518d7fb2196dea0.js:1:317914
    at e.value (7420.9da7a9f1f16a156b.js:1:482745)
    at e.value (7420.9da7a9f1f16a156b.js:1:491641)
    at n.value (7420.9da7a9f1f16a156b.js:1:468626)
    at 819.1518d7fb2196dea0.js:1:317836
    at _app-d66b0f27d628c577.js:302:1448187
    at 819.1518d7fb2196dea0.js:1:317792
    at u2 (framework-0183e6d1d8e5eebe.js:9:92097)
um @ framework-0183e6d1d8e5eebe.js:9
a.componentDidCatch.t.callback @ framework-0183e6d1d8e5eebe.js:9
l5 @ framework-0183e6d1d8e5eebe.js:9
oa @ framework-0183e6d1d8e5eebe.js:9
e @ framework-0183e6d1d8e5eebe.js:9
(anonymous) @ framework-0183e6d1d8e5eebe.js:9
o8 @ framework-0183e6d1d8e5eebe.js:9
oQ @ framework-0183e6d1d8e5eebe.js:9
lf @ framework-0183e6d1d8e5eebe.js:9
oK @ framework-0183e6d1d8e5eebe.js:9
eD @ framework-0183e6d1d8e5eebe.js:9
rx @ framework-0183e6d1d8e5eebe.js:9
nB @ framework-0183e6d1d8e5eebe.js:9
nA @ framework-0183e6d1d8e5eebe.js:9
www-embed-player.js:952 
        
        
       POST https://www.youtube.com/youtubei/v1/log_event?alt=json net::ERR_BLOCKED_BY_CLIENT
zm @ www-embed-player.js:952
Dm @ www-embed-player.js:957
Lr @ www-embed-player.js:1364
(anonymous) @ www-embed-player.js:1272
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
ar @ www-embed-player.js:1271
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:1275
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
(anonymous) @ www-embed-player.js:1275
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
b.options.onError @ www-embed-player.js:1274
(anonymous) @ www-embed-player.js:957
(anonymous) @ www-embed-player.js:924
k @ www-embed-player.js:950
XMLHttpRequest.send
zm @ www-embed-player.js:952
Dm @ www-embed-player.js:957
Lr @ www-embed-player.js:1364
(anonymous) @ www-embed-player.js:1272
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
ar @ www-embed-player.js:1271
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:1275
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
(anonymous) @ www-embed-player.js:1275
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
b.options.onError @ www-embed-player.js:1274
(anonymous) @ www-embed-player.js:957
(anonymous) @ www-embed-player.js:924
k @ www-embed-player.js:950
XMLHttpRequest.send
zm @ www-embed-player.js:952
Dm @ www-embed-player.js:957
Lr @ www-embed-player.js:1364
(anonymous) @ www-embed-player.js:1272
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
ar @ www-embed-player.js:1271
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
Zq @ www-embed-player.js:1259
(anonymous) @ www-embed-player.js:1355
Promise.then
Nr @ www-embed-player.js:1355
(anonymous) @ www-embed-player.js:1371
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
Xr @ www-embed-player.js:1371
(anonymous) @ www-embed-player.js:1943
(anonymous) @ HFwLrI6Z2fc?autoplay=0&mute=0&controls=0&origin=https%3A%2F%2Ftry.fiftyone.ai&playsinline=1&showinfo=0&rel=0&iv_load_policy=3&modestbranding=1&enablejsapi=1&widgetid=1:10
www-embed-player.js:952 
        
        
       POST https://www.youtube.com/youtubei/v1/log_event?alt=json net::ERR_BLOCKED_BY_CLIENT
zm @ www-embed-player.js:952
Dm @ www-embed-player.js:957
Lr @ www-embed-player.js:1364
(anonymous) @ www-embed-player.js:1272
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
ar @ www-embed-player.js:1271
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:1275
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
(anonymous) @ www-embed-player.js:1275
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
b.options.onError @ www-embed-player.js:1274
(anonymous) @ www-embed-player.js:957
(anonymous) @ www-embed-player.js:924
k @ www-embed-player.js:950
XMLHttpRequest.send
zm @ www-embed-player.js:952
Dm @ www-embed-player.js:957
Lr @ www-embed-player.js:1364
(anonymous) @ www-embed-player.js:1272
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
ar @ www-embed-player.js:1271
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:1275
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
(anonymous) @ www-embed-player.js:1275
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
b.options.onError @ www-embed-player.js:1274
(anonymous) @ www-embed-player.js:957
(anonymous) @ www-embed-player.js:924
k @ www-embed-player.js:950
XMLHttpRequest.send
zm @ www-embed-player.js:952
Dm @ www-embed-player.js:957
Lr @ www-embed-player.js:1364
(anonymous) @ www-embed-player.js:1272
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
ar @ www-embed-player.js:1271
(anonymous) @ www-embed-player.js:1269
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
(anonymous) @ www-embed-player.js:1269
(anonymous) @ www-embed-player.js:924
setTimeout
apply @ VM7:4265
qm @ www-embed-player.js:944
bo.Za @ www-embed-player.js:1055
xn.pa @ www-embed-player.js:1014
r.Sb @ www-embed-player.js:1269
Zq @ www-embed-player.js:1259
(anonymous) @ www-embed-player.js:1355
Promise.then
Nr @ www-embed-player.js:1355
(anonymous) @ www-embed-player.js:1371
Fa @ www-embed-player.js:41
Ga.next @ www-embed-player.js:42
b @ www-embed-player.js:46
Promise.then
f @ www-embed-player.js:48
(anonymous) @ www-embed-player.js:49
Ha @ www-embed-player.js:48
A @ www-embed-player.js:50
Xr @ www-embed-player.js:1371
(anonymous) @ www-embed-player.js:1943
(anonymous) @ HFwLrI6Z2fc?autoplay=0&mute=0&controls=0&origin=https%3A%2F%2Ftry.fiftyone.ai&playsinline=1&showinfo=0&rel=0&iv_load_policy=3&modestbranding=1&enablejsapi=1&widgetid=1:10

```

</details>

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

The `ClassificationsOverlay` does not have a `label` attribute. Resolves #5317 

## How is this patch tested? If it is not, please explain why.

Unit test for edge cases

## Release Notes

* Fixed visualizer rendering for classification labels

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function for retrieving array buffers from overlays.
	- Added new overlay types: Detection, Heatmap, and Segmentation.

- **Bug Fixes**
	- Improved logic for handling overlay data to enhance performance and reliability.
	- Updated cleanup method to prevent potential errors.

- **Chores**
	- Adjusted imports for better modularity and organization.
	- Enhanced type safety in utility functions for improved reliability.
	- Added new test suite for the `viewsAreEqual` function to ensure correct behavior.

### CODE REVIEW

File: app/packages/looker/src/lookers/abstract.ts [hunk 1]
```diff
@@ -546,9 +552,9 @@ export abstract class AbstractLooker<
         if (buffer.detached) {
           // most likely sample is already being processed, skip update
           return;
-        } else {
-          arrayBuffers.push(buffer);
         }
+
+        arrayBuffers.push(buffer);
```
Comment (sashankaryal) [inline app/packages/looker/src/lookers/abstract.ts:557]: Should we add a guard for undefined buffer?
```suggestion
        if (buffer) {
          arrayBuffers.push(buffer);
        }
```

Comment (benjaminpkane) [reply app/packages/looker/src/lookers/abstract.ts:557]: There is a falsey check higher up already. I will do a little more cleaning, though
